projects/ng-qrcode/src/lib/qr-code.component.ts

```diff
@@ -46,4 +50,20 @@ export class QrCodeComponent {
   @Input()
   margin?: number
 
+  refreshing = false
+
+  resizeEventListener: EventListener = () => {
+    this.refreshing = true
+    setTimeout(() => {
+      this.refreshing = false
```

Comment:
I think this is the part that I'm most uncomfortable about

There's a few things happening here:
- The resize events aren't being de-bounced, and each event is triggering a new `setTimeout`
- We're using the `refreshing` property to remove the canvas from the DOM and then add it back, which causes the whole thing to flicker noticeably

I can see why you added the `refreshing`, as otherwise the element can only get wider when resizing, since when making the window narrower the canvas is already occupying space and thus prevents the parents width from decreasing.

It looks like this might have also been required in the demo app, in order to trigger the resize to occur when changing the column widths.

I think it's probably possible to achieve the desired behavior using a combination of `position: absolute` (to remove it's size from the parents size calculations) and a [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) on the parent element rather than the window (to avoid needing to force a refresh if the parents width changes but the window width doesn't)

From a template perspective I think it would look something like this:
```
@Component({
  selector: "qr-code",
  template: `
    <div [class.relative]="fillTheParentElement" [style.height.px]="_size">
      <canvas *ngIf="value"
              [qrCode]="value"
              [qrCodeErrorCorrectionLevel]="errorCorrectionLevel"
              [qrCodeCenterImageSrc]="centerImageSrc"
              [qrCodeCenterImageWidth]="centerImageSize"
              [qrCodeCenterImageHeight]="centerImageSize"
              [qrCodeMargin]="margin"
              [width]="_size"
              [height]="_size"
              [darkColor]="darkColor"
              [lightColor]="lightColor"
              [fillTheParentElement]="fillTheParentElement"
              [class.absolute]="fillTheParentElement"
      >
      </canvas>
    </div>
  `,
  styles: [`
    .relative {
      position: relative;
      top: 0;
      left: 0;
    }
    .absolute {
      position: absolute;
      top: 0;
      left: 0;
    }
  `],
})
```

Though I think there are still some major drawbacks to this, and perhaps the flickering isn't so bad.

Reply:
I absolutely agree with you. When I was writing these lines, I felt really bad :( Then I thought "Resizing is not a common event for end users", so I used this method. I'll change ResizeObserver to the parent element from the window, it will be absolutely nice. I'll take a look for this solution, thanks!